### PR TITLE
Add ECMWF metview-batch and magics-metview-batch

### PR DESCRIPTION
**Pull request by iainrussell**
ECMWF's data processing/visualisation package Metview already has a feedstock, but this includes the user interface, which brings in Qt and OpenGL dependencies, meaning that it cannot run on remote systems, as happens for example in Binder notebooks. The proposed metview-batch feedstock is based on the existing metview feedstock, but without any UI dependencies.

It also requires a new variant of ECMWF's Magics package (added here in the same PR as magics-metview-batch) that is built with Metview support, but without Qt support.

I am the maintainer of Metview at ECMWF.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by iainrussell** [comment]
@StephanSiemen - are you willing to be listed as a maintainer of metview-batch and magics-metview-batch?

**Comment by iainrussell** [comment]
@sandorkertesz - are you willing to be listed as a maintainer of metview-batch?

**Comment by iainrussell** [comment]
@sylvielamythepaut - are you willing to be listed as a maintainer of magics-metview-batch?

**Comment by sylvielamythepaut** [comment]
Of course, no problem

**Comment by StephanSiemen** [comment]
> @StephanSiemen - are you willing to be listed as a maintainer of metview-batch and magics-metview-batch?

Yes, of course

**Comment by sandorkertesz** [comment]
Yes, thanks a lot

**Comment by iainrussell** [comment]
Hi @conda-forge/staged-recipes - this PR is ready now. It adds a batch-only version of Metview to be used in environments such as Binder.

File: recipes/magics-metview-batch/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  skip: True  # [win]
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - cmake
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+  host:
+    - pkg-config
```
Comment (scopatz) [inline recipes/magics-metview-batch/meta.yaml]: pkg-config is usually a build dep

File: recipes/magics-metview-batch/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    - boost
```
Comment (scopatz) [inline recipes/magics-metview-batch/meta.yaml]: Can you use `boost-cpp` here?  The `boost` package on conda-forge include boost-python

Comment (scopatz) [reply recipes/magics-metview-batch/meta.yaml]: I guess you are using python here

File: recipes/magics-metview-batch/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    - LICENSE
+
+  license_family: Apache
+  summary: "ECMWF's Meteorological plotting software with support for metview-batch."
+
+extra:
+  recipe-maintainers:
+    - iainrussell
+    - StephanSiemen
+    - sylvielamythepaut
```
Comment (scopatz) [inline recipes/magics-metview-batch/meta.yaml]: @StephanSiemen @sylvielamythepaut are you willing to be maintainers here?

File: recipes/metview-batch/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    - eccodes
+    - magics-metview-batch
+    - boost
+    - flex
+    - bison
+    - libnetcdf
+    - proj
+    - gdbm
+    - fftw
+    - libopenblas
```
Comment (scopatz) [inline recipes/metview-batch/meta.yaml]: Do you rely on openblas specifically?  See https://conda-forge.org/docs/maintainer/knowledge_base.html?#blas

File: recipes/magics-metview-batch/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    - 0001-Find-proj_6_0.lib-if-proj.lib-isn-t-found.patch
+
+build:
+  number: 0
+  skip: True  # [win]
+  detect_binary_files_with_prefix: true
+
+requirements:
+  build:
+    - cmake
```
Comment (scopatz) [inline recipes/magics-metview-batch/meta.yaml]: You should have make here too ;)

**Comment by iainrussell** [comment]
Excellent feedback @scopatz ! I'll look at them all. I already asked the maintainers if they were happy, and they commented 'yes' in this PR  :)

File: recipes/magics-metview-batch/meta.yaml (hunk 1 above)
Comment (iainrussell) [reply recipes/magics-metview-batch/meta.yaml]: @scopatz  - done!

File: recipes/magics-metview-batch/meta.yaml (hunk 2 above)
Comment (iainrussell) [reply recipes/magics-metview-batch/meta.yaml]: @scopatz - in fact, I could use boost-cpp - thanks for the suggestion, it's done now.

File: recipes/metview-batch/meta.yaml (hunk 4 above)
Comment (iainrussell) [reply recipes/metview-batch/meta.yaml]: @scopatz - openblas was added as an experiment to get past a feature that I've now disabled, so I removed it from the deps.

File: recipes/magics-metview-batch/meta.yaml (hunk 5 above)
Comment (iainrussell) [reply recipes/magics-metview-batch/meta.yaml]: @scopatz   :) Done!

**Comment by iainrussell** [comment]
So @scopatz , I believe I've resolved all the issues you raised (thanks for taking the time to look and to make great suggestions), the maintainers have given their blessing and the checks are green! Are you the person to merge the PR?  Thanks again for your help with this!

File: recipes/metview-batch/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    - eccodes
+    - magics-metview-batch
+    - boost
+    - flex
+    - bison
+    - libnetcdf
+    - proj
+    - gdbm
+    - fftw
+    - _openmp_mutex  # [linux]
```
Comment (isuruf) [inline recipes/metview-batch/meta.yaml]: See https://conda-forge.org/docs/maintainer/knowledge_base.html#openmp on how to properly indicate `openmp`.

File: recipes/metview-batch/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  run:
+    - libcurl
+    - eccodes
+    - magics-metview-batch
+    - flex
+    - bison
```
Comment (isuruf) [inline recipes/metview-batch/meta.yaml]: Do you really need `flex` and `bison` at runtime?

**Comment by iainrussell** [comment]
Thanks so much, @scopatz and @chrisburr! Metview users will be very happy :)

File: recipes/metview-batch/meta.yaml (hunk 7 above)
Comment (iainrussell) [reply recipes/metview-batch/meta.yaml]: Hi @isuruf, just noticed this comment - I think you're right, this is only needed at build time. I'll wait for conda-forge and do its magic, then I can make that change in the 'real' repo. Thanks for taking the time to look, and to point that out! I fully agree that we should minimise the run-time dependencies for the users.

**Pull request merged**